File: try/ipy/VirtualAccessors.py
import sys

import clr
clr.AddReference("./VirtualAccessors.dll")
import SetOverride, GetOverride

s = SetOverride()
g = GetOverride()

try:
  s.Accessor = 1
except:
  print >>sys.stderr, "SetOverride setter failed."

try:
  s.Accessor
except:
  print >>sys.stderr, "SetOverride getter failed."

try:
  g.Accessor = 1
except:
  print >>sys.stderr, "GetOverride setter failed."

try:
  g.Accessor
except:
  print >>sys.stderr, "SetOverride getter failed."

File: try/ipy/VirtualAccessors.cs
public class Base {
  public virtual int Accessor {
    get; set;
  }
}

public class GetOverride: Base {
  public override int Accessor {
    get {
      return base.Accessor - 1;
    }
  }
}

public class SetOverride: Base {
  public override int Accessor {
    set {
      base.Accessor = value + 1;
    }
  }
}

